io-engine/src/subsys/config/opts.rs

```diff
@@ -264,58 +316,32 @@ where
     }
 }
 
-impl Default for NvmfTcpTransportOpts {
-    fn default() -> Self {
-        Self {
-            max_queue_depth: try_from_env("NVMF_TCP_MAX_QUEUE_DEPTH", 32),
-            in_capsule_data_size: 4096,
-            max_io_size: 131_072,
-            io_unit_size: 131_072,
-            max_qpairs_per_ctrl: try_from_env(
-                "NVMF_TCP_MAX_QPAIRS_PER_CTRL",
-                32,
-            ),
-            num_shared_buf: try_from_env("NVMF_TCP_NUM_SHARED_BUF", 2047),
-            buf_cache_size: try_from_env("NVMF_TCP_BUF_CACHE_SIZE", 64),
-            dif_insert_or_strip: false,
-            max_aq_depth: 32,
-            abort_timeout_sec: 1,
-            acceptor_poll_rate: try_from_env("NVMF_ACCEPTOR_POLL_RATE", 10_000),
-            zcopy: try_from_env("NVMF_ZCOPY", 1) == 1,
-            ack_timeout: try_from_env("NVMF_ACK_TIMEOUT", 0),
-            data_wr_pool_size: try_from_env("NVMF_DATA_WR_POOL_SIZE", 0),
-        }
-    }
-}
-
 /// we cannot add derives for YAML to these structs directly, so we need to
 /// copy them. The upside though, is that if the FFI structures change, we will
 /// know about it during compile time.
-impl From<NvmfTcpTransportOpts> for spdk_nvmf_transport_opts {
-    fn from(o: NvmfTcpTransportOpts) -> Self {
-        struct_size_init!(
-            Self {
-                max_queue_depth: o.max_queue_depth,
-                max_qpairs_per_ctrlr: o.max_qpairs_per_ctrl,
-                in_capsule_data_size: o.in_capsule_data_size,
-                max_io_size: o.max_io_size,
-                io_unit_size: o.io_unit_size,
-                max_aq_depth: o.max_aq_depth,
-                num_shared_buffers: o.num_shared_buf,
-                buf_cache_size: o.buf_cache_size,
-                dif_insert_or_strip: o.dif_insert_or_strip,
-                reserved29: Default::default(),
-                abort_timeout_sec: o.abort_timeout_sec,
-                association_timeout: 120000,
-                transport_specific: std::ptr::null(),
-                acceptor_poll_rate: o.acceptor_poll_rate,
-                zcopy: o.zcopy,
-                reserved61: Default::default(),
-                ack_timeout: o.ack_timeout,
-                data_wr_pool_size: o.data_wr_pool_size,
-            },
-            opts_size
-        )
+impl From<NvmfTransportOpts> for spdk_nvmf_transport_opts {
+    fn from(o: NvmfTransportOpts) -> Self {
+        Self {
```

Comment:
Why is this not being initialized the same as before with struct_size_init! ?

Reply:
good catch, not sure if this was something prompted to do by build or a rookie mistake. I've set it back to using macro.